pkg/crosscluster/logical/replication_statements.go

```diff
@@ -28,6 +35,27 @@ func getPhysicalColumns(table catalog.TableDescriptor) []catalog.Column {
 	return result
 }
 
+func getPhysicalColumnsSchema(table catalog.TableDescriptor) []columnSchema {
+	columns := getPhysicalColumns(table)
```

Comment:
As you pointed out in that test, this will miss computed columns that are part of the PK.

Reply:
@msbutler pointed out the fact we really do need composite keys in the lookup. Which was a serious miss on my part. Fixing that is going to require some small adjustments throughout the crudy sql writer, so I'll take it on in a follow up PR.